### DISCUSSION TITLE
SXS Issue with Steam Input

### DISCUSSION THREAD

**Discussion by DaRealSparky** [nefarius/DsHidMini#241] [Q&A]
So I go to play a new game with my DS3. I load up default binding in Steam for that game. Everything seems fine initially. Slightest touch of the triggers, BOOM! Constant, non-stop rapid fire of the triggers. Doesn't matter which one I touched. Tried different response curves to no avail. Makes anything that needs or uses triggers unplayable. The only programs that don't have this issue are emulators. (PCSX2, Dolphin, etc.)

How is this fixed or is this a bug with the SXS mode.

**Comment by nefarius** [comment]
[Have you consulted the documentation?](https://vigem.org/projects/DsHidMini/HID-Device-Modes-Explained/) Last time I checked SXS and Steam worked perfectly fine, try other configurations, experiment.

**Reply by DaRealSparky** [reply]
It persists. Another interesting problem, the Playstation/Guide button seems to perform a mouse click action, but only after the triggers start acting up. Disconnecting and reconnecting the controller seems to remedy both issues, but it comes back as soon as you touch a trigger. Just tested with XInput mode and it seems to work fine. I would rather like to be able to use SXS, especially once I get RPCS3 up and running.

**Reply by DaRealSparky** [reply]
Update: While I was testing out steam controller configs and comparing input signals from other controllers, I have developed a theory. (I am by no means an expert, so this could by completely wrong) The axis for both triggers reads in windows as 255, the max analog value. Steam is seeing this and thinking that I am holding the trigger when I have only pressed it once. Testing proves this to be fairly accurate. I also managed to get SDF Mode working in Steam after redefining the controller layout in Gamepad settings in Big Picture Mode. Attempting a similar process in SXS Mode does not seem to work, although it did fix a miss mapping issue where it bound multiple buttons to the trigger. 

Is it possible to invert the trigger values in software? Maybe make it a hot load setting like muting the digital buttons in SDF Mode. Just a thought.

**Reply by DaRealSparky** [reply]
A new idea/question popped up. If the DS3 has pressure sensitive/analog face buttons, wouldn't it be possible to expose those in SXS Mode as well for use in PCSX2? It would make it less work for going between PS2, PS3, and Steam on the same PC.

At this point I'm just spit-balling here. I'm not a coder or anything, but would love to be able to contribute ideas to the future of the project.

**Reply by nefarius** [reply]
All you ask for is already working.
